### ISSUE TITLE
[Test] Move from Karma to Jest

### ISSUE THREAD

**Issue by LeoDog896** [nasa/openmct#7329]
#### Summary

[Karma is deprecated](https://github.com/karma-runner/karma/#karma-is-deprecated-and-is-not-accepting-new-features-or-general-bug-fixes] and heavily recommends either Vitest or Jest. However, as mentioned before, [Vitest isn't quite popular](https://github.com/nasa/openmct/issues/6591#issuecomment-1871368957) compared to Jest, and Jest does have better ESM support (which can help with implementing #6591 later on).

Plus, if Vitest (somehow) becomes the preferred testing framework, [Vitest is primarily compatible with Jest anyway](https://vitest.dev/guide/migration.html#migrating-from-jest).

**Comment by unlikelyzero** [comment]
@LeoDog896 I didn't realize that Karma had been formally deprecated. Looks like the EOL will be some time in 2024.

- I would have said that https://github.com/skovhus/jest-codemods would make the most sense as a migration guide, but in the era of chatgpt, I'm not sure if that's the case.
- We've talked internally about moving to e2e or component tests for our plugins and keep unit tests explicitly for our APIs. This seems like the best time to make that distinction and we could remove a significant number of tests.
- We already combine our unit test and e2e test coverage with codecov https://app.codecov.io/gh/nasa/openmct/flags and likely could identify the plugin coverage that could be removed in the migration. 
- Code coverage would need to be part of the final implementation but not the first.
- I don't mind running jest tests in parallel with karma during the migration.

**Comment by LeoDog896** [comment]
> I would have said that https://github.com/skovhus/jest-codemods would make the most sense as a migration guide, but in the era of chatgpt, I'm not sure if that's the case.
Yep! Used that as my first attempt in migrating it. However, Jest doesn't like webpack, and Jest _especially_ doesn't like usage of `import` in a non `type: module` package. That's the secondary reason as to why I made #7330 and #7331
